chapter-12-platform-specific-apps/desktop-app/src/renderer/apiRequest.ts

```diff
@@ -0,0 +1,50 @@
+import {ApplicationError} from '../shared/applicationError';
+import {IpcEventNames} from '../shared/ipcEventNames';
+import {ApiOptions} from './apiOptions';
+import {ipcRequest} from './ipcRequest';
+
+/*
+ * A generic method in the React app to make reliable API that handle token refresh and expiry
+ */
+export async function makeApiRequest(options: ApiOptions): Promise<any> {
```

Comment:
This is the key function where we require readability to get across the book's message to implement reliable API requests that handle 401s.